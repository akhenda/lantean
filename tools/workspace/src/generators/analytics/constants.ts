/**
 * Default name of the library.
 */
export const defaultLibName = 'analytics';

/**
 * Default directory of the library.
 */
export const defaultLibDirectory = '';

/**
 * Default tags of the library.
 */
export const defaultLibTags = ['web', 'mobile', 'server', 'client', 'analytics'];

/**
 * Dependencies of the library.
 */
export const deps = {
  'expo-application': '^5.9.1',
  'expo-device': '^6.0.2',
  'expo-file-system': '^17.0.1',
  'expo-localization': '^15.0.3',
  'posthog-js': '^1.181.0',
  'posthog-node': '^4.2.1',
  'posthog-react-native': '^3.3.10',
} as const;
