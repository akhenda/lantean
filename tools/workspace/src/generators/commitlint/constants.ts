import { CommitlintConfigFileName } from './types';

export const commitlintConfigFile: CommitlintConfigFileName =
  '.commitlintrc.js';

export const devDependencies = {
  '@commitlint/cli': '^19.4.0',
  '@commitlint/config-conventional': '^19.2.2',
  '@commitlint/config-nx-scopes': '^19.3.1',
  '@theunderscorer/nx-semantic-release': '^2.12.0',
  'cz-git': '^1.9.4',
} as const;
