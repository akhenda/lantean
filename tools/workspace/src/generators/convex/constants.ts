/**
 * Default name of the library.
 */
export const defaultLibName = 'backend';

/**
 * Default directory of the library.
 */
export const defaultLibDirectory = '';

/**
 * Default tags of the library.
 */
export const defaultLibTags = ['server', 'backend', 'type:convex', 'domain:server'];

/**
 * Dependencies of the library.
 */
export const deps = {
  '@edge-runtime/vm': '^4.0.4',
  convex: '^1.17.0',
  'convex-test': '^0.0.34',
} as const;

export const devDeps = {} as const;
