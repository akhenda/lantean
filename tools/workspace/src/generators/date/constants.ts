/**
 * Default name of the library.
 */
export const defaultLibName = 'date';

/**
 * Default directory of the library.
 */
export const defaultLibDirectory = '';

/**
 * Default tags of the library.
 */
export const defaultLibTags = ['web', 'mobile', 'server', 'client', 'date'];

/**
 * Dependencies of the library.
 */
export const deps = {
  'dayjs': '^1.11.13',
} as const;
