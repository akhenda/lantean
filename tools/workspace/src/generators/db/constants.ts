/**
 * Default name of the library.
 */
export const defaultLibName = 'db';

/**
 * Default directory of the library.
 */
export const defaultLibDirectory = '';

/**
 * Default tags of the library.
 */
export const defaultLibTags = ['server', 'db'];

/**
 * Dependencies of the library.
 */
export const deps = {
  '@clerk/nextjs': '^6.3.2',
  '@clerk/themes': '^2.1.43',
  dotenv: '^16.4.5',
  'dotenv-expand': '^11.0.6',
  'drizzle-orm': '^0.36.1',
  'drizzle-zod': '^0.5.1',
  pg: '^8.13.1',
  postgres: '^3.4.5',
  ulid: '^2.3.0',
  zod: '^3.23.8',
} as const;

export const devDeps = {
  'cross-env': '^7.0.3',
  'drizzle-kit': '^0.28.0',
} as const;
