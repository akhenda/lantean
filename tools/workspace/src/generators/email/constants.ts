/**
 * Default name of the library.
 */
export const defaultLibName = 'email';

/**
 * Default directory of the library.
 */
export const defaultLibDirectory = '';

/**
 * Default tags of the library.
 */
export const defaultLibTags = ['server', 'email'];

/**
 * Dependencies of the library.
 */
export const deps = {
  '@react-email/components': '^0.0.27',
  '@react-email/tailwind': '^1.0.1',
} as const;
