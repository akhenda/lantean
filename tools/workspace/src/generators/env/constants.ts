/**
 * Default name of the library.
 */
export const defaultLibName = 'env';

/**
 * Default directory of the library.
 */
export const defaultLibDirectory = 'configs';

/**
 * Default tags of the library.
 */
export const defaultLibTags = [
  'config',
  'configs',
  'constants',
  'web',
  'mobile',
  'server',
  'universal',
  'client',
];

/**
 * Dependencies of the library.
 */
export const envDeps = {
  '@t3-oss/env-core': '^0.11.1',
  zod: '^3.23.8',
} as const;
