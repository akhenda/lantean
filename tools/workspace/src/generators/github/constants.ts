export const installDepsActionPath = './.github/actions/install-deps/actions.yml';
export const verifyConditionsActionPath = './.github/actions/verify-conditions/actions.yml';
export const ciMergeWorkflowPath = './.github/workflows/merge.yml';
