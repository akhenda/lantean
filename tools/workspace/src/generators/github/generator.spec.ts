import { NxJsonConfiguration, readJson, Tree } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import { parse } from 'yaml';

import {
  getGitRepo,
  nxConfigFile,
  readmeFile,
  readPackageJson,
  existsGitHubCiWorkflow,
} from '../../devkit';

import gitHubWorkflowGenerator from './generator';
import { ciMergeWorkflowPath } from './constants';

jest.mock('../../devkit', () => ({
  ...jest.requireActual('../../devkit'),
  getGitRepo: jest.fn(),
}));

describe('@lantean/workspace github workflow generator', () => {
  let tree: Tree;

  beforeEach(() => {
    tree = createTreeWithEmptyWorkspace();
    jest.spyOn(console, 'log').mockImplementation(() => null);
    jest.spyOn(console, 'error').mockImplementation(() => null);
    (getGitRepo as jest.Mock).mockReturnValue('https://github.com/test/test');
  });

  it('should run successfully', async () => {
    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    expect(existsGitHubCiWorkflow(tree, ciMergeWorkflowPath)).toBeTruthy();
  });

  it('should skip execution if a GitHub CI workflow already exists', async () => {
    tree.write(ciMergeWorkflowPath, '');

    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: false,
      packageManager: 'bun',
    });

    expect(console.log).toHaveBeenCalledWith(
      `GitHub workflow already existing at path: ${ciMergeWorkflowPath}`
    );
  });

  it('should not skip execution if a GitHub CI workflow already exists but passing force option', async () => {
    tree.write(ciMergeWorkflowPath, '');

    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    expect(tree.read(ciMergeWorkflowPath)?.toString()).toBeTruthy();
  });

  it('should generate the GitHub CI workflow', async () => {
    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    const ci = parse(tree.read(ciMergeWorkflowPath)?.toString() ?? '');

    expect(ci).toMatchSnapshot();
  });

  it('should declare the nx script in package.json', async () => {
    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    const packageJson = readPackageJson(tree);

    expect(packageJson.scripts?.nx).toBe('nx');
  });

  it('should declare the implicit dependency in nx.json', async () => {
    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    const nxConfig = readJson<NxJsonConfiguration>(tree, nxConfigFile);

    expect(nxConfig.namedInputs).toStrictEqual({
      ci: ['{workspaceRoot}/.github/workflows/*.yml'],
      default: ['ci'],
    });
  });

  it('should add a badge in readme', async () => {
    tree.write(readmeFile, '# Readme\n');
    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    const readme = tree.read(readmeFile)?.toString() ?? '';

    expect(readme).toContain('[![CI]');
  });

  it('should not add a badge in readme if repo can not be resolved', async () => {
    (getGitRepo as jest.Mock).mockReturnValue(null);
    tree.write(readmeFile, '# Readme\n');
    await gitHubWorkflowGenerator(tree, {
      branch: 'main',
      useNxCloud: true,
      force: true,
      packageManager: 'bun',
    });

    const readme = tree.read(readmeFile)?.toString() ?? '';

    expect(readme).not.toContain('[![CI]');
    expect(console.error).toHaveBeenCalledWith(
      `Could not add badge to README, remote repo could not be detected.`
    );
  });
});
