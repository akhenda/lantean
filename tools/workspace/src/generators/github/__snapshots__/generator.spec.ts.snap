// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lantean/workspace github workflow generator should generate the GitHub CI workflow 1`] = `
{
  "jobs": {
    "ci-coverage-badge": {
      "name": "Coverage Badge",
      "needs": "ci-validate",
      "secrets": "inherit",
      "uses": "./.github/workflows/ci-coverage-badge.yml",
    },
    "ci-finale": {
      "if": "always()
&& contains(needs.*.result, 'success')
&& !contains(needs.*.result, 'failure')
",
      "name": "Finale",
      "needs": [
        "ci-coverage-badge",
        "ci-release",
      ],
      "uses": "./.github/workflows/ci-finale.yml",
    },
    "ci-nx-init": {
      "name": "NX Cloud Init",
      "uses": "./.github/workflows/ci-nx-init.yml",
    },
    "ci-release": {
      "name": "Release",
      "needs": "ci-validate",
      "secrets": "inherit",
      "uses": "./.github/workflows/ci-release.yml",
    },
    "ci-repo-integrity": {
      "name": "Repo Integrity",
      "needs": "ci-nx-init",
      "uses": "./.github/workflows/ci-repo-integrity.yml",
    },
    "ci-validate": {
      "name": "Validate (Lint/Test/Build)",
      "needs": "ci-repo-integrity",
      "uses": "./.github/workflows/ci-validate.yml",
    },
  },
  "name": "Merge",
  "on": {
    "push": {
      "branches": [
        "main",
        "dev",
      ],
    },
  },
}
`;
